### PR TITLE
SLP wallet creation

### PR DESCRIPTION
**Pull request by fyookball**

### CODE REVIEW

File: gui/qt/installwizard.py [hunk 1]
```diff
@@ -462,27 +483,32 @@ def waiting_dialog(self, task, msg):
         t.join()
 
     @wizard_dialog
-    def choice_dialog(self, title, message, choices, run_next, extra_button=None):
+    def choice_dialog(self, title, message, choices, run_next, extra_button=None,extra_checkbox=None,slp_selection=False):
```
Comment (markblundeberg) [inline gui/qt/installwizard.py]: This `slp_selection` argument breaks the abstraction here. Why does every choice_dialog need to have this special hack thingy that only possibly applies to the "new" dialog? You're better off making it so the return value includes all relevant information.

File: lib/wallet.py [hunk 2]
```diff
@@ -2878,13 +2885,16 @@ def is_multisig(self):
         return True
 
 
-wallet_types = ['standard', 'multisig', 'imported']
+
+#slptest_ wallet types to be renamed later before making available.  They are dev only/obscure wallet type.
```
Comment (markblundeberg) [inline lib/wallet.py]: I'd suggest not merging anything until the actual names have been settled on. Likewise for the 'dev-only' switch above, these just say to me the code is still a work in progress.

Comment (fyookball) [reply lib/wallet.py]: @markblundeberg the idea is be able to merge into mainline without maintaining a separate branch and without disrupting the main UX.  Maybe that is not best practice but it works and is low overhead.  Right now no one else is making SLP merge happen.

**Comment by fyookball** [comment]
Mark suggested maybe we want to do wallet flag instead of different wallet type.  Actually sounds logical to me.  Thoughts @markblundeberg @cculianu

**Comment by fyookball** [comment]
closing this.  Will resubmit later using wallet flag rather than new wallet type.

**Comment by cculianu** [comment]
We really really really just need to have me sit down and do dual-path.  Kicking that can down the road with boolean flags and whatnot is not going to make anything easier.

Users should have 1 wallet that can potentially "see" an SLP path and manage it.  
SLP isn't going away and all HD wallets should be SLP capable.

User can opt to not see SLP UI if they wish -- but in my mind they should be able to OPT IN to SLP at any time they wish (esp. if someone just sent them an SLP token!!).

At that point the wallet needs to monitor the 245 path.

This involves some code changes to wallet.py -- which is a huge file but I have stared at it so much in the past 6 months i can even tell you where in the file all the various bits are.  Like I have a mental map of that file.

It is a job that needs to be done eventually.  UX for users should just be "I got a wallet".  

And that wallet should "speak" SLP if the user wants. 

It's not that hard.  Requires me or someone with the devotion to sit down and spend many days on it ironing out the code.

@imaginaryusername Agrees with me on this.

Doing anything but dual path is kicking the can down the road and creating "technical debt" (which I know is a trigger word now so I hesitate to use it).

Thanks.

**Pull request closed without merging**